<?xml version="1.0" ?>
<robot name = "robot_arm">

<!--base link-->
<link name = "base_link" />

<!-- link 1 -->
<link name = "link1">
    <visual>
        <geometry>
            <box size="1 0.2 0.08"/>
        </geometry>
        <origin xyz="0 0 0" rpy="0 0 0"/>
    </visual>

     <collision>
        <geometry>
            <box size="0.35 0.09 0.09"/>
        </geometry>
        <origin xyz="0 0 0" rpy="0 0 0"/>
    </collision>

    <inertial>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <mass value="1"/>
        <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.01"/>
    </inertial>
</link>
<!--joint1-->
<joint name="joint1" type="fixed">
    <parent link="base_link"/>
    <child link="link1"/>
    <origin xyz="0 0 0" rpy="0 0 0"/>
    <axis xyz="0 0 0"/>
    <limit lower="0" upper="0" effort="100" velocity="1"/>
</joint>

<!-- link 2 -->
<link name = "link2">
    <visual>
        <geometry>
            <cylinder length="0.1" radius="0.05"/>
        </geometry>
        <origin xyz="0 0 0" rpy="0 0 0"/>
    </visual>

     <collision>
        <geometry>
            <cylinder length="0.1" radius="0.05"/>
        </geometry>
        <origin xyz="0 0 0" rpy="0 0 0"/>
    </collision>

    <inertial>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <mass value="1"/>
        <inertia ixx="0" ixy="0" ixz="0" iyy="0.001" iyz="0" izz="0"/>
    </inertial>
</link>
<!--joint2-->
<joint name="joint2" type="prismatic">
    <parent link="link1"/>
    <child link="link2"/>
    <origin xyz="-0.4 0 0.09" rpy="0 0 0"/> 
    <axis xyz="1 0 0"/>
    <limit lower="0" upper="-0.83" effort="100" velocity="1"/>
</joint>

<!-- link 3 -->
<link name = "link3">
    <visual>
        <geometry>
            <box size="0.1 1 0.06"/>
        </geometry>
        <origin xyz="0 0 0" rpy="0 0 0"/>
    </visual>

     <collision>
        <geometry>
            <box size="0.35 0.09 0.09"/>
        </geometry>
        <origin xyz="0.15 0 0" rpy="0 0 0"/>
    </collision>

    <inertial>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <mass value="1"/>
        <inertia ixx="0.01" ixy="0" ixz="0" iyy="0.01" iyz="0" izz="0.01"/>
    </inertial>
</link>

<!--joint3-->
<joint name="joint3" type="revolute">
    <parent link="link2"/>
    <child link="link3"/>
    <origin xyz="0 0 0.08" rpy="0 0 0"/> 
    <axis xyz="0 0 1"/>
    <limit lower="0" upper="180" effort="100" velocity="1"/>
</joint>

</robot>
